hybrasyl/Objects/Creature.cs

```diff
@@ -109,12 +221,272 @@ public Creature()
             Gold = 0;
             Inventory = new Inventory(59);
             Equipment = new Inventory(18);
+            BaseDamageModifier = 1;
+            BonusDamageModifier = 0;
+            BaseHealModifier = 1;
+            BonusHealModifier = 0;
+            BaseReflectIntensity = 1;
+            BaseReflectChance = 0;
+            DamageTypeOverride = null;
+            Condition = new ConditionInfo(this);            
         }
 
         public override void OnClick(User invoker)
         {
         }
 
+        public Creature GetDirectionalTarget(Direction direction)
+        {
+            VisibleObject obj;
+
+            switch (direction)
+            {
+                case Direction.East:
+                    {
+                        obj = Map.EntityTree.FirstOrDefault(x => x.X == X + 1 && x.Y == Y);
+                    }
+                    break;
+                case Direction.West:
+                    {
+                        obj = Map.EntityTree.FirstOrDefault(x => x.X == X - 1 && x.Y == Y);
+                    }
+                    break;
+                case Direction.North:
+                    {
+                        obj = Map.EntityTree.FirstOrDefault(x => x.X == X && x.Y == Y - 1);
+                    }
+                    break;
+                case Direction.South:
+                    {
+                        obj = Map.EntityTree.FirstOrDefault(x => x.X == X && x.Y == Y + 1);
+                    }
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(direction), direction, null);
+            }
+            if (obj is Creature) return obj as Creature;
+            return null;
+        }
+    
+        public virtual List<Creature> GetTargets(Castable castable, Creature target = null)
+        {
```

Comment:
An array would likely be better in this instance

Reply:
Why?

Reply:
In the case of a list desyncing due to creature spawn or death. Could be me overthinking it, but I can see the list being modified, and erroring. An array would not have this issue.